src/client/src/orangehrmRecruitmentPlugin/components/CandidateProfile.vue

```diff
@@ -0,0 +1,283 @@
+import FileUploadInput from '@/core/components/inputs/FileUploadInput';
+import DateInput from '@/core/components/inputs/DateInput';
+import {navigate} from '@/core/util/helper/navigation';
+export default {
+  name: 'CandidateProfile',
+  components: {
+    DateInput,
+    'vacancy-dropdown': VacancyDropdown,
+    'file-upload-input': FileUploadInput,
+    'full-name-input': FullNameInput,
+  },
+  props: {
+    candidateId: {
+      type: Number,
+      required: true,
+    },
+  },
+  emits: ['getData'],
+  setup(props) {
+    const http = new APIService(
+      'https://c81c3149-4936-41d9-ab3d-e25f1bff2934.mock.pstmn.io',
+      `/recruitment/candidate/${props.candidateId}`,
+    );
+
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      allowedFileTypes: [
+        'application/pdf',
+        'application/msword',
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+        'text/plain',
+        'application/rtf',
+        'application/vnd.oasis.opendocument.text',
+      ],
+      profile: {
+        firstName: '',
+        middleName: '',
+        lastName: '',
+        email: '',
+        contactNumber: '',
+        oldResume: '',
+        notes: '',
+        keywords: '',
+        newResume: null,
+        vacancy: null,
+        resume: null,
+        method: 'keepCurrent',
+        applicationDate: null,
+        keep: null,
+      },
+      rules: {
+        firstName: [required, shouldNotExceedCharLength(30)],
+        lastName: [required, shouldNotExceedCharLength(30)],
+        middleName: [shouldNotExceedCharLength(30)],
+        email: [required, validEmailFormat, shouldNotExceedCharLength(50)],
+        contactNumber: [validPhoneNumberFormat, shouldNotExceedCharLength(25)],
+        keywords: [shouldNotExceedCharLength(250)],
+        applicationDate: [validDateFormat()],
+        resume: [
+          maxFileSize(1024 * 1024),
+          validFileTypes(this.allowedFileTypes),
+        ],
+      },
+    };
+  },
+
+  beforeMount() {
+    this.isLoading = true;
+    this.http.getAll().then(({data: {data}}) => {
+      const {resume, candidate, manager, ...rest} = data;
+      const managerName = manager.terminationId
+        ? '(Past Employee)'
+        : '' + `${manager.firstName} ${manager.middleName} ${manager.lastName}`;
+      const fullName = `${candidate.firstName} ${candidate.middleName} ${candidate.lastName}`;
+      this.profile.oldResume = resume?.id ? resume : null;
+      this.profile.newResume = null;
+      this.profile.firstName = candidate.firstName;
+      this.profile.middleName = candidate.middleName;
+      this.profile.lastName = candidate.lastName;
+      this.profile.method = 'keepCurrent';
+      this.profile.vacancy = data.vacancy;
+      this.profile = {
+        ...this.profile,
+        ...rest,
+      };
+      this.isLoading = false;
+      this.$emit('getData', {
```

Comment:
Your **CandidateProfileLayout** approach is better than previous method. But i don't like to share state like this by events. it honestly looks fragile. either we can take this common state out to `CandidateProfileLayout.vue` component or we should individually get state in `CandidateProfile.vue` and `RecruitmentStatus.vue` from two APIs.

By doing later i feel you can bypass even more code duplication. the reason you created `HiredAction.vue`, `RejectAction.vue,` `JobOfferedAction.vue` etc is because depending on the status the action defers. Imagine if `RecruitmentStatus.vue` had it own API, to get the status and the actions possible to perform. Let's talk more

Reply:
just add another API